src/components/transactions/SingleTx/index.tsx

```diff
@@ -1,59 +1,50 @@
-import { useRouter } from 'next/router'
 import useSafeInfo from '@/hooks/useSafeInfo'
-import useAsync from '@/hooks/useAsync'
-import type { DateLabel, Transaction, TransactionDetails } from '@gnosis.pm/safe-react-gateway-sdk'
-import { getTransactionDetails } from '@gnosis.pm/safe-react-gateway-sdk'
-import { sameAddress } from '@/utils/addresses'
+import type { Transaction, TransactionDetails } from '@gnosis.pm/safe-react-gateway-sdk'
 import type { ReactElement } from 'react'
-import { makeDateLabelFromTx, makeTxFromDetails } from '@/utils/transactions'
+import { makeTxFromDetails } from '@/utils/transactions'
 import { TxListGrid } from '@/components/transactions/TxList'
-import TxDateLabel from '@/components/transactions/TxDateLabel'
 import ExpandableTransactionItem from '@/components/transactions/TxListItem/ExpandableTransactionItem'
+import { useTxDetails } from '@/hooks/useTxDetails'
+import { isMultisigDetailedExecutionInfo } from '@/utils/transaction-guards'
+import { GroupLabelTypography, useFutureNonceLabel } from '../GroupLabel'
 
 const SingleTxGrid = ({ txDetails }: { txDetails: TransactionDetails }): ReactElement => {
   const tx: Transaction = makeTxFromDetails(txDetails)
-  const dateLabel: DateLabel = makeDateLabelFromTx(tx)
 
   return (
     <TxListGrid>
-      <TxDateLabel item={dateLabel} />
       <ExpandableTransactionItem item={tx} txDetails={txDetails} testId="single-tx" />
     </TxListGrid>
   )
 }
 
 const SingleTx = () => {
-  const chainId = useChainId()
-  const router = useRouter()
-  const { id } = router.query
-  const transactionId = Array.isArray(id) ? id[0] : id
-  const { safe, safeAddress } = useSafeInfo()
-
-  const [txDetails, txDetailsError] = useAsync<TransactionDetails>(
-    () => {
-      if (!transactionId) return
-      return getTransactionDetails(chainId, transactionId)
-    },
-    [transactionId, safe.txQueuedTag, safe.txHistoryTag],
-    false,
-  )
-  const isCurrentSafeTx = sameAddress(txDetails?.safeAddress, safeAddress)
+  const [txDetails, error] = useTxDetails()
+  const futureNonceLabel = useFutureNonceLabel()
+
+  const { safe } = useSafeInfo()
 
-  const error = !transactionId
-    ? new Error("Couldn't retrieve the transaction details. Please review the URL.")
-    : !isCurrentSafeTx
-    ? new Error(`Transaction with id ${transactionId} not found in this Safe`)
-    : txDetailsError
+  let nonceWarning: string | undefined = undefined
+  const executionInfo = txDetails?.detailedExecutionInfo
+  if (isMultisigDetailedExecutionInfo(executionInfo)) {
+    if (executionInfo.nonce > safe.nonce) {
+      nonceWarning = futureNonceLabel
+    }
+  }
 
   if (error) {
     return <ErrorMessage error={error}>Failed to load transaction</ErrorMessage>
   }
 
   if (txDetails) {
-    return <SingleTxGrid txDetails={txDetails} />
+    return (
+      <Box display="flex" flexDirection="column" gap={1}>
+        {nonceWarning && <GroupLabelTypography label={nonceWarning} />}
```

Comment:
Maybe we should show the `DateLabel` if there is no nonce warning, wdyt?

Reply:
I asked Lila and she said it's not necessary as we show the executionDate in the Accordion.